src/components/KTableView/KTableView.vue

```diff
@@ -229,18 +229,18 @@
 
           <tbody>
             <template
-              v-for="(row, rowIndex) in tableData"
-              :key="`table-${tableId}-row-${rowIndex}`"
+              v-for="(row, rowIndex) in data"
+              :key="`table-${tableId}-row-${getRowKey(row) ?? useUniqueId()}`"
```

Comment:
Yes I think it should be stable within the same table which only backend API can guarantee. Generating keys on the frontend should always have its issues, either you are generating different keys for the same object (randomizing), or same keys for different object (using index).

Reply:
> What about using a WeakMap to hold the generated keys for the row objects that didn't change, so that we won't be reconstructing the rows at least when the same row objects are used here?

If I'm understanding you correctly, you suggest that we create a `WeakMap` that will hold all generated keys for each row (we'd still use `useUniqueId` function to make the keys unique if `rowKey` is not provided). Then in the `v-for` we reference the corresponding key from the `WeakMap` for each row? If so then I don't quite understand how would we benefit from this approach.

Scenario: `rowKey` is not provided, pagination page size is 15 so the table renders with 15 unique keys in `WeakMap`. User navigates to page 2 of table data which also contains 15 records so we need to generate new values for each row. Then user returns to page 1. We would have no way of identifying the keys that were generated for page 1 the first time the table was rendered since the keys don't contain a unique identifier associated with values sourced from data objects. So we would have to generate new keys for each row - effectively what we are doing right now.

Reply:
The `WeakMap` stores row object => generated key. We only generate new keys if different row objects are passed in. If I understand correctly, at lease we are not generating new keys if the re-render comes from props other than `data`, or the new `data` contains some of the row objects as in the previous value?

/cc @adamdehaven WDYT?

Reply:
> at lease we are not generating new keys if the re-render comes from props other than data

Yes that’s what I was thinking

Reply:
Okay, let me try that

Reply:
Implemented ([commit](https://github.com/Kong/kongponents/pull/2427/commits/493320adb4c52f865e3e4da49d92226ae296fbf5)), please review.